### ISSUE TITLE
Since 0.10.6, programatic links do not work anymore

### ISSUE THREAD

**Issue by geomagilles** [kwebio/kweb-core#218]
**Describe the bug**
This code :
```
     val a = a().setAttribute("href", "#")
         
       a.on.click {
            println("clicked")
        }
```
works (display "clicked" when clicking on the a element) for 0.10.5 - not anymore for 0.10.6 (seems to route to # instead)

**Comment by sanity** [comment]
Hmm, I haven't looked into this yet, but does it work ok if you don't do the setAttribute()?

**Comment by geomagilles** [comment]
Yes it works without the href. 
I add that having a href set to # seems a common practice to me (used by other frameworks) - without it the browser does not identify the element as a link

**Comment by sanity** [comment]
@geomagilles I'm having trouble reproducing this problem, may I ask which browser you're using?

**Comment by sanity** [comment]
FYI - in the meantime, I'm adding a property to AElement, so you can do something like:
```kotlin
a().href = "/foobar"
```
For a relative URL like this (beginning with `/`) it will override the default click behavior and modify `WebBrowser.url` instead, avoiding a page refresh. For other URLs it will just set the `href` attribute.

**Comment by geomagilles** [comment]
I'm using Safari - but the issue seems a bit more complex than that (>0.10.6) : when I change the url by `WebBrowser.url.value = ...`, it works, but instantly come back to the previous url.
I'm looking at my code to see if something can explain this behavior

**Comment by geomagilles** [comment]
I have the feeling that the on.click works, but in parallel the link to "#" is processed, triggering a back to the initial url.

**Comment by geomagilles** [comment]
Btw, I apologize: my initial description of the bug was incorrect. I've tried to simplify the description but then I misrepresented it

**Comment by geomagilles** [comment]
comparing to Vue (https://vuejs.org/v2/guide/events.html), one possible syntax to avoid this could be:
- `a.on.click.prevent { }`

Another possible option is to consider that the on.click presence implies to not follow the link

**Comment by geomagilles** [comment]
@sanity can you please clarify when the function inside `route { path() {...} }` are called? 
Also it's not clear to me what happens when I change Webbrowser.url

**Comment by sanity** [comment]
@geomagilles The functions inside `route {}` are called on initial page render, and are based on the value of `WebBrowser.url` - if `url` changes then they are re-rendered automatically.

The idea is to be able to change the page URL without a full page refresh. Does that answer your question?

**Comment by geomagilles** [comment]
@sanity yes, but I'm not sure now about the difference between a page refresh and updating url (as in your new a().href syntax)

**Comment by sanity** [comment]
@geomagilles A page refresh would download the HTML again re-rendering the whole page from scratch, while just changing `WebBrowser.url` will just change the page URL and the parts of the DOM that need to change, so it's a lot faster.

See [here](https://smartlazycoding.com/javascript-tutorial/change-the-url-without-reloading-the-page) for a more detailed explanation.

**Comment by sanity** [comment]
Should be fixed in `0.10.9` if you use AElement.href to set the hyperlink because it will override the default click behavior. Please reopen if there are any issues.

**Comment by geomagilles** [comment]
Thx - seems ok to me.
